registry/server/templates/routes/getRenderedTemplate.ts

```diff
@@ -28,13 +29,30 @@ const validateRequestBeforeGetTemplateRendered = validateRequestFactory([{
 }]);
 
 async function getRenderedTemplate(req: Request<GetTemplateRenderedRequestParams>, res: Response): Promise<void> {
+    let defaultTemplate;
+
     const {
         name: templateName,
     } = req.params;
 
-    const { locale } = req.query;
+    const { locale, domain } = req.query;
+
+    if (domain) {
+        const [domainItem] = await db.select('id').from<RouterDomains>('router_domains').where('domainName', String(domain));
+
+        if (domainItem) {
+            [defaultTemplate] = await db.select('templates.*').from<Template>('templates')
+                .join('routes', 'templates.name', 'routes.templateName')
+                .where({
+                    domainId: domainItem.id,
+                    name: templateName,
+                });
+        }
+    }
 
-    const [defaultTemplate] = await db.select().from<Template>('templates').where('name', templateName);
+    if (!defaultTemplate) {
```

Comment:
at line 57 we have the same condition and it's confusing. What do you think about creating different variables, e.g. `templateByDomain` and `defaultTemplate`?
or at least somehow rewrite it, since it's really hard to read when you see the code the first time.